app/javascript/esbuild.js

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env node
+const fs = require('fs')
+const ImportGlobPlugin = require('esbuild-plugin-import-glob')
+const browserslistToEsbuild = require('browserslist-to-esbuild')
+
+function build() {
+  const env = require('./.config/env.json')
+  require('esbuild')
+    .build({
+      entryPoints: [
+        './app/javascript/packs/application.tsx',
+        './app/javascript/packs/core.tsx',
+        './app/javascript/packs/internal.tsx',
+        ...(process.env.RAILS_ENV === 'test'
+          ? ['./app/javascript/packs/test.tsx']
+          : []),
+      ],
+      bundle: true,
+      sourcemap: true,
+      format: 'esm',
+      splitting: true,
+      minify: process.env.NODE_ENV === 'production',
+      watch: process.argv.includes('--watch'),
+      outdir: '.built-assets',
+      tsconfig: './tsconfig.json',
+      target: browserslistToEsbuild(),
+      define: {
+        // TODO: move bugsnag API key into config
+        'process.env.BUGSNAG_API_KEY': '"938ae3d231c5455e5c6597de1b1467af"',
+        'process.env.WEBSITE_ASSETS_HOST': `"${env['website_assets_host']}"`,
```

Comment:
Our of interest do we need the nested string and interpolation thing here?

Reply:
We do, as esbuild will interpret the value as JSON.

Changing the first option to `'process.env.BUGSNAG_API_KEY': '938ae3d231c5455e5c6597de1b1467af'` results in the following error:

```
$ ./app/javascript/esbuild.js
✘ [ERROR] Invalid define value (must be valid JSON syntax or a single identifier): /assets
```

Reply:
Great. Thanks.